### PR TITLE
Add support for TOML files to the CLI.

### PR DESCRIPTION
**Pull request by Julian**
Oh hello there :)

Do you have any interest in TOML support for the CLI? If so, here's a PR -- where I've basically used `tomllib` on 3.11 and tomli before.

If you have other preferences on the below obviously feel free to lemme know, or to close if it isn't in scope!

### CODE REVIEW

File: docs/cli.rst [hunk 1]
```diff
@@ -22,8 +22,9 @@ All the power of ``glom``, without even opening your text editor!
 
    --help / -h                     show this help message and exit
    --target-file TARGET_FILE       path to target data source (optional)
-   --target-format TARGET_FORMAT   format of the source data (json or python)
-                                     (defaults to 'json')
+   --target-format TARGET_FORMAT
+                                   format of the source data (json, python, toml
```
Comment (Julian) [inline docs/cli.rst:26]: (Even if for whatever reason we don't merge this, this was missing `yaml` in the `--help`)

Comment (mahmoud) [reply docs/cli.rst:26]: Good catch! I'm an oxford comma fan, so I'll probably just self-apply this suggestion:

```suggestion
                                   format of the source data (json, python, toml,
```

File: glom/cli.py [hunk 2]
```diff
@@ -11,8 +11,8 @@
   --help / -h                 show this help message and exit
   --target-file TARGET_FILE   path to target data source (optional)
   --target-format TARGET_FORMAT
-                              format of the source data (json or python)
-                              (defaults to 'json')
+                              format of the source data (json, python, toml
```
Comment (mahmoud) [inline glom/cli.py:14]:
```suggestion
                              format of the source data (json, python, toml,
```

File: glom/cli.py [hunk 3]
```diff
@@ -111,7 +111,7 @@ def mw_handle_target(target_text, target_format):
 
     Args:
         target_text (str): String that specifies where 6
-        target_format (str): Valid formats include `.json` and `.yml` or `.yaml`
+        target_format (str): Valid formats include `.json`, `.toml` and `.yml` or `.yaml`
```
Comment (mahmoud) [inline glom/cli.py:114]:
```suggestion
        target_format (str): Valid formats include `.json`, `.toml`, and `.yml` or `.yaml`
```

File: glom/cli.py [hunk 4]
```diff
@@ -111,7 +111,7 @@ def mw_handle_target(target_text, target_format):
 
     Args:
         target_text (str): String that specifies where 6
```
Comment (mahmoud) [inline glom/cli.py:113]:
```suggestion
        target_text (str): The target data to load, as text
```

Comment (mahmoud) [reply glom/cli.py:113]: Dunno what happened to the docstring here once upon a time.

**Review by mahmoud — APPROVED** [review]
Hey, wow, this is great! Pretty much perfect, in fact. I've added some Oxford commas and spotted a couple other docs items I'll address post-merge. Thanks so much!

**Pull request merged**